src/EditorFeatures/CSharpTest/MoveToNamespace/MoveToNamespaceTests.cs

```diff
@@ -0,0 +1,451 @@
+﻿// Copyright(c) Microsoft.All Rights Reserved.Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.MoveToNamespace;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities.MoveToNamespace;
+using Microsoft.VisualStudio.Composition;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.MoveToNamespace
+{
+    public class MoveToNamespaceTests : AbstractCSharpCodeActionTest
```

Comment:
Are there tests for cancellation somewhere? Different styles of new namespace names, ones that introduce conflicts that must be resolved in other files, etc. etc.? Or is that tested elsewhere?

Reply:
Cancellation and some various namespace styles I'll add. Conflict resolution should be handled by `IChangeNamespaceService`